### ISSUE TITLE
inconsistent handling of async request failures

### ISSUE THREAD

**Issue by marina-p** [microsoft/restler-fuzzer#178]
In the network.testing logs, the PUT storageAccount (line 111) returned a 202 Accepted (line 148), then there were multiple GET requests (lines 151 to 177) for polling the created resource [...], which ended with (line 179)
2021-04-01 22:14:43.375: Resource polling timed out before the resource was available.
- yet the matching   speccov.json file states that this PUT request was ‘valid’ (line 143) because of the “202 Accepted” (lines 149-150)

So this is a RESTler bug: RESTler is inconsistent and should either declare “202 Accepted” as valid or invalid, but not valid when tested in isolation (see above) and invalid when replayed to get to a child request (the case below).

**Comment by PatGod** [comment]
Later on, RESTler did find an interesting bug after successfully generating such a resource 'storageAccount'.

Note that if we had declared that first PUT request as invalid, then RESTler would have never tried to explore further, and it would not have found that 500 bug!…

So perhaps a '202 Accepted' should still be considered valid, or 'partly valid', in order to let the analysis proceed to deeper resources. Or a new 'error type' should be created in speccov.json (instead of the currently misleading "invalid due to request sequence".

In other words, the design of the solution should be discussed further...

**Comment by PatGod** [comment]
Here is a way to solve this issue, without reducing API coverage and without preventing RESTler to explore deeper:

1. RESTler keeps considering "202 Accepted" as "valid" for API-coverage metric (no change here).

2. when failing to create a child resource because of "202 Accepted"+"failed GET polling requests", we introduce a new category "invalid_due_to_unvailable_parent_resource" in speccov.json

Another solution is to do nothing and keep RESTler as-is: we keep reporting the cases (2) with the existing "invalid_due_to_sequence_failure" (without distinguishing this particular case).
